extensions/azurecore/src/azureResource/commands.ts

```diff
@@ -181,17 +181,24 @@ export function registerAzureResourceCommands(appContext: AppContext, azureViewT
 		vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
 	});
 
-	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode) => {
+	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode | ObjectExplorerActionsContext) => {
 		if (!node) {
 			return;
 		}
+		let connectionProfile: azdata.IConnectionProfile = null;
 
-		const treeItem: azdata.TreeItem = await node.getTreeItem();
-		if (!treeItem.payload) {
-			return;
+		if (node instanceof TreeNode) {
+			const treeItem: azdata.TreeItem = await node.getTreeItem();
+			if (!treeItem.payload) {
+				return;
+			}
+			// Ensure connection is saved to the Connections list, then open connection dialog
+			connectionProfile = Object.assign({}, treeItem.payload, { saveProfile: true });
```

Comment:
We don't want to do this logic  for the OE context as well?

Reply:
this is just for extensions